tree/treeplayer/src/TTreeProcessorMT.cxx

```diff
@@ -91,32 +91,141 @@ std::vector<std::vector<Long64_t>> GetFriendEntries(const std::vector<std::pair<
 }
 }
 
+////////////////////////////////////////////////////////////////////////////////
+/// Get and store the names, aliases and file names of the friends of the tree.
+/// \param[in] tree The main tree whose friends to 
+Internal::FriendInfo TTreeProcessorMT::GetFriendInfo(TTree &tree)
+{
+   std::vector<Internal::NameAlias> friendNames;
+   std::vector<std::vector<std::string>> friendFileNames;
+
+   const auto friends = tree.GetListOfFriends();
+   if (!friends)
+      return Internal::FriendInfo{};
+
+   for (auto fr : *friends) {
+      const auto frTree = static_cast<TFriendElement *>(fr)->GetTree();
+
+      // Check if friend tree has an alias
+      const auto realName = frTree->GetName();
+      const auto alias = tree.GetFriendAlias(frTree);
+      if (alias) {
+         friendNames.emplace_back(std::make_pair(realName, std::string(alias)));
+      } else {
+         friendNames.emplace_back(std::make_pair(realName, ""));
+      }
+
+      // Store the file names of the friend tree
+      friendFileNames.emplace_back();
+      auto &fileNames = friendFileNames.back();
+      const bool isChain = tree.IsA() == TClassRef("TChain");
+      if (isChain) {
+         const auto frChain = static_cast<TChain *>(frTree);
+         for (auto f : *(frChain->GetListOfFiles())) {
+            fileNames.emplace_back(f->GetTitle());
+         }
+      } else {
+         const auto f = frTree->GetCurrentFile();
+         fileNames.emplace_back(f->GetName());
+      }
+   }
+
+   return Internal::FriendInfo{std::move(friendNames), std::move(friendFileNames)};
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// Retrieve the name of the first TTree in the first input file, else throw.
+std::string TTreeProcessorMT::GetTreeName()
```

Comment:
Can we call this method `FindTreeName`? Otherwise at first sight it might seem a getter of `fTreeName`.

Reply:
sure